### PR TITLE
Fixed static fields initialization in WorkflowStepFactory

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Fixed static fields initialization. Pulling the `Supplier<WorkflowStep>` inside the enum over complicates the initialization of the static fields. Reverting the stepMap initialization inside the `WorkflowStepFactory` constructor.
Also, removed the unused `clusterService` and `client` passed.

### Issues Resolved
Fixes https://github.com/opensearch-project/flow-framework/issues/531

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — APPROVED** [review]
Code changes LGTM.  Change log may need tweaking.

**Review by amitgalitz — APPROVED** [review]
LGTM, thanks for fixing this

**Review by joshpalis — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.1.0/)
 ### Features
 ### Enhancements
 ### Bug Fixes
+- Fixed static fields initialization in WorkflowStepFactory ([#532](https://github.com/opensearch-project/flow-framework/pull/532))
```
[the remark this answers is not in the document]
Comment (amitgalitz) [reply CHANGELOG.md]: everything is tracked through https://github.com/opensearch-project/anomaly-detection/blob/main/.github/workflows/draft-release-notes-workflow.yml, during release we just copy paste into release notes and its done

[the remark this answers is not in the document]
Comment (amitgalitz) [reply CHANGELOG.md]: this also makes it easy to do the github releases

[the remark this answers is not in the document]
Comment (owaiskazi19) [reply CHANGELOG.md]: Changelog is the way we do it in core as well.

**Pull request merged**